feature/platform/transceiver/tests/zr_input_output_power_test/zr_input_output_power_test.go

```diff
@@ -0,0 +1,416 @@
+package zr_input_output_power_test
+
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package topology_test configures just the ports on DUT and ATE,
+// assuming that DUT port i is connected to ATE i.  It detects the
+// number of ports in the testbed and can be used with the 2, 4, 12
+// port variants of the atedut testbed.
+
+import (
+	"context"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/featureprofiles/internal/samplestream"
+	gnps "github.com/openconfig/gnoi/system"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/testt"
+	"github.com/openconfig/ygot/ygot"
+)
+
+const (
+	opticalChannelTransceiverType = oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_OPTICAL_CHANNEL
+	maxRebootTime                 = 900
+	maxCompWaitTime               = 600
+	samplingTime                  = 10000000000 //10 Seconds
+	targetOutputPower             = -9
+	interfaceFlapTimeOut          = 30 //seconds
+	outputFreqLowerBound          = 184500000
+	outputFreqUpperBound          = 196000000
+	rxSignalPowerLowerBound       = -14
+	rxSignalPowerUpperBound       = 0
+	txOutputPowerLowerBound       = -10
+	txOutputPowerUpperBound       = -6
+)
+
+type testData struct {
+	transceiverName               string
+	dut                           *ondatra.DUTDevice
+	transceiverOpticalChannelName string
+	interfaceName                 string
+}
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// configureDUT removes any breakoutConfiguration if present, and configure DUT
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice, transceiverName string) {
+	d := gnmi.OC()
+	port := dut.Port(t, "port1")
+
+	//remove any breakout configuration
```

Comment:
Go style says single line comments must:
- have a space after //
- begin with a capital letter
- end with a period

Reply:
Addressed in the latest commit.

Reply:
Resolved